trieste/acquisition/optimizer.py

```diff
@@ -192,73 +194,182 @@ def optimize_continuous(
+        leading dimension of `starting_points` controls the number of individual
+        optimization runs.
+    :param optimizer_args: Keyword arguments to pass to the Scipy optimizer.
+    :return: A list containing a Scipy OptimizeResult object for each optimization.
+    """
+
+    tf_dtype = starting_points.dtype  # type for communication with Trieste
+
+    num_optimization_runs = tf.shape(starting_points)[0].numpy()
+
+    def _objective_value(x: TensorType) -> TensorType:
+        return -target_func(tf.expand_dims(x, 1))  # [len(x),1]
+
+    def _objective_value_and_gradient(x: TensorType) -> Tuple[TensorType, TensorType]:
+        return tfp.math.value_and_gradient(_objective_value, x)  # [len(x), 1], [len(x), D]
+
+    if isinstance(
+        space, TaggedProductSearchSpace
+    ):  # build continuous relaxation of discrete subspaces
+        bounds = [
+            get_bounds_of_box_relaxation_around_point(space, starting_points[i : i + 1])
+            for i in tf.range(num_optimization_runs)
+        ]
+    else:
+        bounds = [spo.Bounds(space.lower, space.upper)] * num_optimization_runs
+
+    # Initialize the numpy arrays to be passed to the greenlets
+    np_batch_x = np.zeros((num_optimization_runs, tf.shape(starting_points)[1]), dtype=np.float64)
+    np_batch_y = np.zeros((num_optimization_runs,), dtype=np.float64)
+    np_batch_dy_dx = np.zeros(
+        (num_optimization_runs, tf.shape(starting_points)[1]), dtype=np.float64
+    )
+
+    # Set up child greenlets
+    child_greenlets = [ScipyLbfgsBGreenlet() for _ in range(num_optimization_runs)]
+    child_results: List[Union[spo.OptimizeResult, np.ndarray]] = [
+        gr.switch(starting_points[i].numpy(), bounds[i], optimizer_args)
+        for i, gr in enumerate(child_greenlets)
+    ]
+
+    while True:
+        all_done = True
+        for i, result in enumerate(child_results):  # Process results from children.
+            if isinstance(result, spo.OptimizeResult):
+                continue  # children return a `spo.OptimizeResult` if they are finished
+            all_done = False
+            assert isinstance(result, np.ndarray)  # or an `np.ndarray` with the query `x` otherwise
+            np_batch_x[i, :] = result
+
+        if all_done:
+            break
+
+        # Batch evaluate query `x`s from all children.
+        batch_x = tf.constant(np_batch_x, dtype=tf_dtype)  # [num_optimization_runs, d]
+        batch_y, batch_dy_dx = _objective_value_and_gradient(batch_x)
+        np_batch_y = batch_y.numpy().astype("float64")
+        np_batch_dy_dx = batch_dy_dx.numpy().astype("float64")
+
+        for i, greenlet in enumerate(child_greenlets):  # Feed `y` and `dy_dx` back to children.
+            if greenlet.dead:  # Allow for crashed greenlets
+                continue
+            child_results[i] = greenlet.switch(np_batch_y[i], np_batch_dy_dx[i, :])
+
+    return child_results
+
+
+class ScipyLbfgsBGreenlet(gr.greenlet):
```

Comment:
Can greenlets fall with exceptions? Do they catch them all and show "deal == True"? If not, shall we be catching any outside this greenlet?